src/linalg/udu.rs

```diff
@@ -0,0 +1,78 @@
+#[cfg(feature = "serde-serialize")]
+use serde::{Deserialize, Serialize};
+
+use crate::allocator::Allocator;
+use crate::base::{DefaultAllocator, MatrixN, VectorN, U1};
+use crate::dimension::Dim;
```

Comment:
This is needed for one of my other suggestions:
```suggestion
use crate::dimension::Dim;
use crate::storage::Storage;
```